src/traces/isosurface/attributes.js

```diff
@@ -0,0 +1,184 @@
+/**
+* Copyright 2012-2018, Plotly, Inc.
+* All rights reserved.
+*
+* This source code is licensed under the MIT license found in the
+* LICENSE file in the root directory of this source tree.
+*/
+
+'use strict';
+
+var colorscaleAttrs = require('../../components/colorscale/attributes');
+var colorbarAttrs = require('../../components/colorbar/attributes');
+var surfaceAtts = require('../surface/attributes');
+var baseAttrs = require('../../plots/attributes');
+
+var extendFlat = require('../../lib/extend').extendFlat;
+
+module.exports = extendFlat({
+    x: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the X coordinates of the vertices on X axis.'
+        ].join(' ')
+    },
+    y: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Y coordinates of the vertices on Y axis.'
+        ].join(' ')
+    },
+    z: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc+clearAxisTypes',
+        description: [
+            'Sets the Z coordinates of the vertices on Z axis.'
+        ].join(' ')
+    },
+    volume: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        description: [
+            'Sets the 4th dimension of the vertices.'
+        ].join(' ')
+    },
+    isocap: {
+        valType: 'boolean',
+        role: 'info',
+        editType: 'calc',
+        dflt: false,
+        description: [
+            'Enable caps between minimum and maximum iso values.'
+        ].join(' ')
+    },
+    isovalue: {
+        valType: 'data_array',
+        role: 'info',
+        editType: 'calc',
+        dflt: [0],
+        description: [
+            'Sets iso surface boundaries.'
+        ].join(' ')
+    },
+    meshalgo: {
```

Comment:
I'd vote for removing this attribute from first version. Using marching cubes appears sufficient for now - unless @archmoj can convince me otherwise.

Reply:
The `SurfaceNets` algorithm is capable of creating smoother isosurfaces. The only issue with that is the end points are not reaching the cube boundaries. Anyway that may be a good option for the user in particular scenarios. Example below presents the mock with `SurfaceNets`.
![isosurface_using_surfacenets](https://user-images.githubusercontent.com/33888540/50254541-6719f480-03bc-11e9-88b1-6b47e2079fbd.png)

Reply:
Done in https://github.com/plotly/plotly.js/pull/3340/commits/dd08cd463f2f6f9ace4fbe5da44748fef21ad2f8 .